src/components/pages/SingleAccommodation.js

```diff
@@ -33,35 +42,85 @@ export class SingleAccommodation extends React.Component {
     this.setState({ isLoading: true });
     await this.props.GetSingleAccommodation(slug);
     this.setState({ isLoading: false });
-    this.single = this.singleAccommodation.bind(this);
+    this.single = this.renderAcommodation.bind(this);
+    this.props.getBookings();
   }
 
-  renderAcommodation() {
-    const { accommodation } = this.props;
-    const { userId } = this.state;
-    const post = accommodation;
-    const { ownerUser } = post;
-    const ratingNumber = post.ratings ? post.ratings.length : 0;
-    const imageArray = post.images ? post.images : [];
-    const ratingArray = post.ratings ? post.ratings : [];
-    let images = [];
-
-    if (accommodation) {
-      switch (typeof imageArray) {
-        case 'string':
-          images.push(post.images);
-          break;
-        default:
-          images = imageArray;
-          break;
-      }
-      return (
-        <div key={post.id} className="container-fluid border single-container">
-          <Row className="spaced-row">
-            <Col md={5} className="breadcrumbs">
-              <Breadcrumbs itemsArray={['> Home', '  accommodations', post.name]} />
-            </Col>
-            {
+  async componentWillUnmount() {
+    const { hideAlert } = this.props;
+    await hideAlert();
+  }
+
+  handleChange = (e) => {
+    const { name, value } = e.target;
+
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  handleCheckIn = async (checkInDate) => {
+    const checkInValue = dateFnsFormat(checkInDate, 'yyyy-MM-dd');
+    await this.setState((state) => ({ ...state, checkInDate: checkInValue }));
+  }
+
+  handleCheckOut = async (checkOutDate) => {
+    const checkOutValue = dateFnsFormat(checkOutDate, 'yyyy-MM-dd');
+    await this.setState((state) => ({ ...state, checkOutDate: checkOutValue }));
+  }
+
+
+handleSubmit = async (e) => {
+  e.preventDefault();
+  const { props, state } = this;
+  const accomodationId = props.accommodation.id;
+  this.setState({ isLoading: true });
+  await props.BookAccommodation(state.checkInDate, state.checkOutDate, accomodationId, state.roomsNumber);
+  this.setState({
+    isLoading: false, checkInDate: '', checkOutDate: '', roomsNumber: '',
+  });
+}
+
+renderAcommodation() {
```

Comment:
This method is too long kindly create a separate component for bookings and call it in this component